packages/SwingSet/src/liveslots/virtualObjectManager.js

```diff
@@ -487,11 +540,40 @@ export function makeVirtualObjectManager(
       if (initializing) {
         cache.remember(innerSelf);
       }
-      const representative = Far(tag, actualize(wrappedData));
+      const self = actualize(wrappedData);
+      let representative;
+      let exposedRepresentative;
+      const facetiousness = assessFacetiousness(self);
+      switch (facetiousness) {
+        case 'one': {
+          representative = Far(tag, self);
+          vrm.checkOrAcquireFacetNames(kindID, null);
+          exposedRepresentative = representative;
+          break;
+        }
+        case 'many': {
+          exposedRepresentative = {};
+          representative = [];
```

Comment:
I'm surprised that `representative` is an array. Given the contrasting names, may I assume this array is never exposed?

On the exposed representative, curious why you don't use `self` itself as you do for the single facet case above. It doesn't matter so no objection. But if it's otherwise indifferent, it would reduce the cost of making the multi-facetted representative.

Reply:
Correct, this array is never exposed.  We use an array to make vref lookup more direct; otherwise we'd also have to store a name table.